server/rest/src/test/java/org/infinispan/rest/resources/ProtobufResourceTest.java

```diff
@@ -0,0 +1,209 @@
+
+      // Read
+      response = client.newRequest(BASE_URL + "/person")
+            .method(HttpMethod.GET)
+            .accept(TEXT_PLAIN_TYPE).send();
+
+      ResponseAssertion.assertThat(response).isOk();
+      ResponseAssertion.assertThat(response).hasContentEqualToFile("person.proto");
+
+      // Read adding .proto should also work
+      response = client.newRequest(BASE_URL + "/person.proto")
+            .method(HttpMethod.GET)
+            .accept(TEXT_PLAIN_TYPE).send();
+      ResponseAssertion.assertThat(response).isOk();
+      ResponseAssertion.assertThat(response).hasContentEqualToFile("person.proto");
+
+      // Update
+      response = updateSchema("person", personProto);
+      ResponseAssertion.assertThat(response).isOk();
+
+      // Delete
+      response = client.newRequest(BASE_URL + "/person")
+            .content(new StringContentProvider(personProto))
+            .method(HttpMethod.DELETE).send();
+
+      ResponseAssertion.assertThat(response).isOk();
+
+      response = client.newRequest(BASE_URL + "/person")
+            .content(new StringContentProvider(personProto))
+            .method(HttpMethod.GET)
+            .accept(TEXT_PLAIN_TYPE).send();
+
+      ResponseAssertion.assertThat(response).isNotFound();
+   }
+
+   @Test
+   public void createTwiceSchema() throws Exception {
+      String personProto = getResourceAsString("person.proto", getClass().getClassLoader());
+      ContentResponse response = addSchema("person", personProto);
+      ResponseAssertion.assertThat(response).isOk();
+      response = addSchema("person", personProto);
+      ResponseAssertion.assertThat(response).isConflicted();
+   }
+
+   @Test
+   public void addAndGetListOrderedByName() throws Exception {
+      String personProto = getResourceAsString("person.proto", getClass().getClassLoader());
+      addSchema("users", personProto);
+      addSchema("people", personProto);
+      addSchema("dancers", personProto);
+
+      String url = String.format("http://localhost:%d/rest/v2/schemas", restServer().getPort());
+      ContentResponse response = client.newRequest(url)
+            .method(HttpMethod.GET)
+            .accept(APPLICATION_JSON_TYPE).send();
+
+      ResponseAssertion.assertThat(response).isOk();
+      ObjectMapper objectMapper = new ObjectMapper();
+      JsonNode jsonNode = objectMapper.readTree(response.getContent());
+      assertEquals(3, jsonNode.size());
+      assertEquals("dancers.proto", jsonNode.get(0).get("name").asText());
+      assertEquals("people.proto", jsonNode.get(1).get("name").asText());
+      assertEquals("users.proto", jsonNode.get(2).get("name").asText());
+   }
+
+   private ContentResponse addSchema(String schemaName, String schemaContent)
+         throws InterruptedException, TimeoutException, ExecutionException {
+      return callAPI(schemaName, schemaContent, HttpMethod.POST);
+   }
+
+   private ContentResponse updateSchema(String schemaName, String schemaContent)
```

Comment:
Should change the name of this method (and `updateNonExistingSchema`) considering that `PUT` now means "create or update". I'd just replace `update` with `put`.

Reply:
ok

Reply:
I believe this was missed, however I won't stop this from integrating it.